core/src/test/java/org/elasticsearch/gateway/GatewayIndexStateIT.java

```diff
@@ -277,66 +277,6 @@ public void testTwoNodesSingleDoc() throws Exception {
         }
     }
 
-    public void testDanglingIndicesConflictWithAlias() throws Exception {
-        logger.info("--> starting two nodes");
-        internalCluster().startNodesAsync(2).get();
-
-        logger.info("--> indexing a simple document");
-        client().prepareIndex("test", "type1", "1").setSource("field1", "value1").setRefresh(true).execute().actionGet();
-
-        logger.info("--> waiting for green status");
-        ensureGreen();
-
-        logger.info("--> verify 1 doc in the index");
-        for (int i = 0; i < 10; i++) {
-            assertHitCount(client().prepareSearch().setQuery(matchAllQuery()).get(), 1L);
-        }
-        assertThat(client().prepareGet("test", "type1", "1").execute().actionGet().isExists(), equalTo(true));
-
-        internalCluster().stopRandomNonMasterNode();
-
-        // wait for master to processed node left (so delete won't timeout waiting for it)
-        assertFalse(client().admin().cluster().prepareHealth().setWaitForNodes("1").get().isTimedOut());
-
-        logger.info("--> deleting index");
-        assertAcked(client().admin().indices().prepareDelete("test"));
-
-        index("test2", "type1", "2", "{}");
-
-        logger.info("--> creating index with an alias");
-        assertAcked(client().admin().indices().prepareAliases().addAlias("test2", "test"));
-
-        logger.info("--> starting node back up");
-        internalCluster().startNode();
-
-        ensureGreen();
-
-        // make sure that any other events were processed
-        assertFalse(client().admin().cluster().prepareHealth().setWaitForRelocatingShards(0).setWaitForEvents(Priority.LANGUID).get()
-            .isTimedOut());
-
-        logger.info("--> verify we read the right thing through alias");
-        assertThat(client().prepareGet("test", "type1", "2").execute().actionGet().isExists(), equalTo(true));
-
-        logger.info("--> deleting alias");
-        assertAcked(client().admin().indices().prepareAliases().removeAlias("test2", "test"));
-
-        logger.info("--> waiting for dangling index to be imported");
-
-        assertBusy(new Runnable() {
-            @Override
-            public void run() {
-                assertTrue(client().admin().indices().prepareExists("test").execute().actionGet().isExists());
-            }
-        });
-
-        ensureGreen();
-
-        logger.info("--> verifying dangling index contains doc");
-
-        assertThat(client().prepareGet("test", "type1", "1").execute().actionGet().isExists(), equalTo(true));
-    }
-
```

Comment:
@bleskes The `IndicesClusterStateService#clusterChanged` (which is where deleted indices are applied) gets called before `GatewayMetaState#clusterChanged` (which is where process dangling indices happens).  Now that we have the tombstones specifying an exact name/uuid index to delete, those tombstones will get processed and indices on disk will get deleted before any attempts at dangling indices.  Its not possible for a dangling index with the same name as an alias to be laying around at the import dangling index phase unless the tombstones got wiped out from the cluster state, because that same index would have had to be deleted before assigning an alias to another index with the same name.  So the only way it seems possible to have a dangling index that conflicts with an alias name is the following scenario:

1. Master `M` and data node `D` are in the cluster.
2. An index named `test` is created.
3. `D` goes offline.
4. `M` deletes index `test`.
5. `M` goes offline and has its data directory wiped out.
6. `M` restarts with no indices, no tombstones, and a brand new cluster UUID.
7. `M` creates index `test2` with alias `test`.
8. `D` starts back up, at which point it receives cluster state from master indicating there are no index tombstones (so nothing to delete), and when it tries to import the dangling indices, there is a naming conflict with the alias and it should fail.  However, if we drop the alias, then we should be able to import the dangling index.

So perhaps thats the scenario I should test for?

This scenario also made me realize that we still have the same issue described in #16358 because if the CS gets wiped out and master no longer has the index tombstone for a shadow replica index, then the node that went offline will still try to import the dangling shadow replica index when it restarts, and encounter the same exception in trying to access the deleted shard data.